finality-aleph/src/party/mod.rs

```diff
@@ -324,275 +545,201 @@ mod tests {
-            "`validator_session_started` mismatch"
-        );
-
-        assert_eq!(
-            *controller
-                .node_session_manager
-                .session_early_started
-                .lock()
-                .unwrap(),
-            HashSet::from_iter(early_started),
-            "`session_early_started` mismatch"
-        );
-
-        assert_eq!(
-            *controller
-                .node_session_manager
-                .session_stopped
-                .lock()
-                .unwrap(),
-            HashSet::from_iter(stopped),
-            "`session_stopped` mismatch"
-        );
-
-        assert_eq!(
-            *controller
-                .node_session_manager
-                .nonvalidator_session_started
-                .lock()
-                .unwrap(),
-            HashSet::from_iter(non_validator_started),
-            "`nonvalidator_session_started` mismatch"
-        );
-    }
-
     #[tokio::test(flavor = "multi_thread")]
     async fn party_starts_session_for_node_in_authorities() {
-        let (party, mut controller) = create_mocked_consensus_party();
-
-        controller
-            .node_session_manager
-            .set_node_id(Some(UintAuthorityId(0).to_public_key()));
-
-        let _party_handle = tokio::spawn(party.run());
+        let (test, party) = PartyTest::new(SessionPeriod(SESSION_PERIOD));
 
         let authorities: Vec<_> = (0..10)
             .map(|id| UintAuthorityId(id).to_public_key())
             .collect();
-        set_authorities_for_sessions(
-            &mut controller,
-            vec![
-                (SessionId(0), authorities.clone()),
-                (SessionId(1), authorities),
-            ],
-        )
-        .await;
 
-        sleep(Duration::from_millis(100)).await;
-        simulate_n_blocks(0, SESSION_PERIOD, &controller);
-        sleep(Duration::from_millis(
-            SESSION_STATUS_CHECK_PERIOD.as_millis() as u64 + 100,
-        ))
-        .await;
+        let state_1 = PartyState {
+            validator_started: vec![SessionId(0)],
+            early_started: vec![SessionId(1)],
+            stopped: vec![],
+            non_validator_started: vec![],
+        };
 
-        assert_session_states(
-            &controller,
-            vec![SessionId(0), SessionId(1)],
-            vec![SessionId(1)],
-            vec![SessionId(0)],
-            vec![],
-        );
+        let state_2 = PartyState {
+            validator_started: vec![SessionId(0), SessionId(1)],
+            early_started: vec![SessionId(1)],
+            stopped: vec![SessionId(0)],
+            non_validator_started: vec![],
+        };
+
+        test.set_authorities_for_session_at_block(0, authorities.clone(), SessionId(0))
+            .set_authorities_for_session_at_block(25, authorities, SessionId(1))
+            .set_node_id_for_session_at_block(0, Some(UintAuthorityId(0).to_public_key()))
+            .assert_session_states_at_block(28, state_1)
```

Comment:
Name nitpick: I find the `expect` phrase better here since at the stage of calling `assert_session_states_at_block` we are not asserting anything. Also, `set_node_id_for_session_at_block` has already input args such as `expected_state` etc

Reply:
done